### DISCUSSION TITLE
Peacock's problem

### DISCUSSION THREAD

**Discussion by tmewhy** [idaholab/moose#18048] [Q&A Getting Started] [answered]
Hi all,
I installed moose on ubuntu20.04 **offline** using the official document solution. After I finished the installation, I used a simple case for testing. When opening peacock, the following reminder appeared: 

> QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'   QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'   QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'   QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'   QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'   QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255' 

When I use peacock to open the output file (.e), peacock crashes with the following error:

> QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'   QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'   QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'   QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'   QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'   QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255   
Segmentation fault (core dumped) 

My guess is that qt or vtk has not been installed well, but I have reinstalled them several times and it still has no effect. Thanks for the help!

**Comment by milljm** [comment, marked as answer]
It's impressive you were able to get Peacock to launch at all using the Offline instructions :) I am not seeing anything about Peacock and its myriad amount of dependencies one needs to install in order to make it work.

How did you manage (just curious)? Did you build each from source, or use a package manager?

Indeed this feels like a python binding to one of the VTK or Qt libraries. Probably VTK. Since Peacock at the very least opens without more than a few notices. If it were Qt and it's stack, I would recon Peacock wouldn't load up at all.

We can use 'chigger' to test pieces of your install. Try the following simple test:

```bash
export MOOSE_DIR=/path/to/moose
export PYTHONPATH=/path/to/moose/python:$PYTHONPATH
# where /path/to/moose is your absolute path to moose

cd /path/to/moose/python/chigger/tests/simple
./simple.py
```
The `simple.py` script should run for a bit, and produce a file: `simple.png` (if all goes well). It probably won't in your case, and I am hoping for a bit more information other than `Seg fault`. Regardless, this `moose/python/chigger/tests/` location is littered with tests we can run to try and zero in on the issue. @aeslaughter would know best (tagging if he would like chime in).

**Reply by tmewhy** [reply]
Thanks, I used "apt-offline" and "pip download" to obtain the relevant installation package on an online computer, and then install it on an offline computer. It did take some time : ) Although it has passed the `make -j 6 `test, I am not sure that I installed it correctly.
I ran `simple.py`and got the following prompt: 

> Aborted (core dumped)

**Reply by GiudGiud** [reply]
Is this still an issue? Thinking of closing, we dont have too many resources on peacock these days. 
To get a backtrace on a seg fault in python, should look at https://docs.python.org/3/library/faulthandler.html

**Reply by jianqixi** [reply]
Did you solve the problem? I also met the same issue of "Aborted (core dumped)" when call peacock

**Reply by milljm** [reply]
@jianqixi, what OS are you operating from?

The following has nothing to do with your current situation. But I feel I must mention:

I am currently _removing_ support for Peacock in our next large update to our Conda packages: https://github.com/idaholab/moose/pull/21324

I am opting instead to create a separate package (lets say: [moose-peacock](https://github.com/idaholab/moose/issues/21382)) which will most likely need to be installed into a separate environment e.g. something like this:

(this doesn't work/exist yet, this is just for demonstration purposes)
```bash
mamba create -n moose moose-libmesh moose-tools    # For compiling moose-based application
mamba create -n peacock moose-peacock              # Strictly for running Peacock
```
You would then activate which ever environment you needed at the time.

The reason for separating them is due to the increasingly difficult-to-solve constraints the dependency chain required for Peacock operation V's the dependency chain required for the compilers/libraries necessary to build/run MOOSE based application.

**Reply by milljm** [reply]
Posting this more for myself;

Peacock (on MacOS) requires these Conda packages:

```pre
mamba create -n peacock pyqt vtk matplotlib pandas python=3.x  # (where x is whatever
                                                                  version of python that
                                                                  your moose environment
                                                                  is using)
```
